MaxText/checkpointing.py

```diff
@@ -82,44 +82,47 @@ def load_state_if_possible(checkpoint_manager: CheckpointManager,
      At most one will be non-None. Both can be None if neither checkpoint is
      set.
   """
-  if checkpoint_manager is None:
-    max_logging.log("no checkpoint manager, not restoring checkpoint")
-    return None, None
   def map_to_pspec(data, pspec):
     if isinstance(data, (jax.Array, jax.ShapeDtypeStruct)) \
           and pspec is not None:
       return type_handlers.ArrayRestoreArgs(mesh=mesh, mesh_axes=pspec)
     else:
       return type_handlers.RestoreArgs()
-
   restore_args = jax.tree_util.tree_map(map_to_pspec,
                                         abstract_unboxed_pre_state,
                                         state_mesh_annotations)
-  latest_step = checkpoint_manager.latest_step()
-  if latest_step is not None:
-    max_logging.log(f"restoring state from this run's directory latest step \
-        {latest_step}")
-    return checkpoint_manager.restore(latest_step, abstract_unboxed_pre_state,
-                                      {"restore_args" : restore_args}), None
-  elif first_checkpoint_path != "":
-    max_logging.log(f"restoring state from first_checkpoint_path {first_checkpoint_path}")
-    p = epath.Path(first_checkpoint_path)
-    checkpointer = Checkpointer(checkpoint.PyTreeCheckpointHandler())
-    return None, checkpointer.restore(p,
-                                      item=abstract_unboxed_pre_state,
-                                      restore_args=restore_args).params
-  elif load_from_other_directory != "":
-    p = epath.Path(load_from_other_directory)
-    checkpointer_loader = Checkpointer(checkpoint.PyTreeCheckpointHandler())
-    mngr_loader = CheckpointManager(p, checkpointer_loader, options=CheckpointManagerOptions(create=True))
-    if load_from_other_directory_step == -1:
-      step = mngr_loader.latest_step()
-      max_logging.log(f"restoring state from {load_from_other_directory} latest step {step}")
+
+  if checkpoint_manager is not None:
+    max_logging.log("checkpoint manager exists so trying to load this run's existing checkpoint")
+
+    latest_step = checkpoint_manager.latest_step()
+    if latest_step is not None:
+      max_logging.log(f"restoring state from this run's directory latest step \
+          {latest_step}")
+      return checkpoint_manager.restore(latest_step, abstract_unboxed_pre_state,
+                                        {"restore_args" : restore_args}), None
     else:
-      step = load_from_other_directory_step
-      max_logging.log(f"restoring state from {load_from_other_directory} step {step}")
-    return mngr_loader.restore(step, abstract_unboxed_pre_state,
-                                      {"restore_args" : restore_args}), None
+      max_logging.log("failed to find preexisting checkpoint for this run")
+
+  if load_parameters_from_path != "":
```

Comment:
Maybe we only have checkpoint_path as variable for inference and decode scripts. Another parameter checkpoint_converter_output_path for any conversion scripts.

Additionally code can pass checkpoint_type as an enum to distinguish between the various types of checkpoints and the read/write code can handle accordingly.

Reply:
Agreed checkpoint type as an enum will be a good idea if the complexity keeps rising. But you're not really commenting on changes in this CR right?

Reply:
Yes can be part of a subsequent change.